### PR TITLE
fix: resolve issues ora-00972:identifier is too long

### LINKED ISSUE #5067: ORA-00972: identifier is too long on joins with FindOptions
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[x] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

It's a copy of #693, cause the issue was closed, but the error still exists - at least with oracle.

When I'm using one of the `getRepository(Entity).find({})` functions with realations array set in the `FindOptions`, I'm getting the `too long identifier` error. I think the main reason it doesn't work is because of Oracle, but I cannot be sure about it and I'm not able to change my db.. 

So it would be really cool, if anyone could make a PR, that allows the user to add an alias for `RelationOptions` or (the easier way) let the aliases only contain out of 1-2 letters.

### PR DESCRIPTION
**Pull request by imnotjames**
Use index instead of parameter name for make it shorter alias name.

Closes #6591
Fixes #5067

### CODE REVIEW

**Comment by imnotjames** [comment]
This is a copy of #6591 that's been rebased against `master` so our oracle tests run against it. :tada:

**Comment by imnotjames** [comment]
Something feels wrong - the tests passed WAY too fast..

**Comment by imnotjames** [comment]
https://app.circleci.com/pipelines/github/imnotjames/typeorm/300/workflows/106c389c-cfa0-4485-9bea-2754a85358a0/jobs/1622

File: test/github-issues/5067/issue-5067.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import "reflect-metadata";
+import {expect} from "chai";
+import {Connection} from "../../../src";
+import {closeTestingConnections, createTestingConnections} from "../../utils/test-utils";
+import {OracleDriver} from "../../../src/driver/oracle/OracleDriver";
+
+describe.only("github issues > #5067 ORA-00972: identifier is too long", () => {
```
Comment (imnotjames) [inline test/github-issues/5067/issue-5067.ts:7]: Aha, `describe.only` was set.  I'll undo that.

**Comment by imnotjames** [comment]
Ok - the test had `.only` in it.  I've removed that & also updated the test to set the enabled drivers properly.

**Comment by imnotjames** [comment]
> 1277 passing (9m)
That's much better.

Huzzah!

File: test/github-issues/5067/issue-5067.ts (hunk 1 above)
Comment (MuratGundes) [reply test/github-issues/5067/issue-5067.ts]: Sorry, I forgot to remove after running the tests :/

Comment (imnotjames) [reply test/github-issues/5067/issue-5067.ts]: No worries! :) It happens.

**Comment by pleerock** [comment]
thank you a lot!

**Comment by MuratGundes** [comment]
Thanks a lot for your effort guys!

**Pull request merged**